File: docs/scripts/players/BeginnerPlayer.js
import AbstractPlayer from "./AbstractPlayer.js";

export default class BeginnerPlayer extends AbstractPlayer {
    constructor(grid) {
        super(grid);
    }
    play() {
        let cell = this.getRandomCell(this.getRandomInt(0, this.rows * this.cols - 1));
        if (!cell) {
            cell = this.getRandomCell();
        }
        return cell;
    }
}

File: src/main/java/model/players/BeginnerPlayer.java
package model.players;

import model.Cell;
import model.CellState;

public class BeginnerPlayer extends AbstractPlayer {
    public BeginnerPlayer(CellState[][] grid) {
        super(grid);
    }

    public Cell play() {
        var cell = this.getRandomCell(this.getRandomInt(0, this.rows * this.cols - 1));
        if (cell.isEmpty()) {
            cell = this.getRandomCell();
        }
        return cell.get();
    }
}
